pkg/api/meta.go

```diff
@@ -46,5 +46,13 @@ func ObjectMetaFor(obj runtime.Object) (*ObjectMeta, error) {
 	if err := runtime.FieldPtr(v, "ObjectMeta", &objectMeta); err != nil {
 		return nil, err
 	}
+
+	if objectMeta.Labels == nil {
```

Comment:
I'm uncomfortable with this - nil labels and nil annotations is a valid go-ism - making it here doesn't make this easier to consume.